### PR TITLE
Allows cancellation of synchronizer.read requests

### PR DESCRIPTION
**Pull request by EthanStandel**
* Fixes TSLint errors.
* Replaces shareReplay with publishReplay and refCount because shareReplay was failing to cancel requests on synchronizer.read calls.
* Bumps version to 0.2.1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "prebuild:js": "yarn test",
     "build:js": "node ./scripts/build.js",
     "pretest": "node ./scripts/pretest.js",
-    "test": "node ./scripts/test.js"
+    "test": "node ./scripts/test.js",
+    "lint": "./node_modules/.bin/tslint --project ."
```
Comment (lVlyke) [inline package.json:30]: Thanks for adding this as a task. It would be great if this was also called in the `prebuild:js` task so that this is run each time a new build is compiled to help make sure Lint checks happen before each build.

File: src/state-selector.ts [hunk 2]
```diff
@@ -181,7 +175,8 @@ export class StateSelector<T> {
                         }),
                         tap(() => this.clearPropertyUpdater(propertyName, pendingRequest$)), // Remove the pending request
                         mergeMap(() => this.state$.pipe(take(1))), // Get the newly updated Session
-                        shareReplay(1)
+                        publishReplay(1),
+                        refCount()
```
Comment (lVlyke) [inline src/state-selector.ts:179]: I'm not sure why this change was required to make pending requests cancelable (theoretically they should have been even with `shareReplay(1)`), but I've tested this and confirmed it works.

**Review by lVlyke — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (EthanStandel) [reply package.json:30]: 👍Added it to the prebuild script

**Review by lVlyke — APPROVED** [review]
